app/updatemanager.go

```diff
@@ -125,9 +138,137 @@ func (u *UpdateManager) run(ctx context.Context) error {
 	return nil
 }
 
+type UpdateControlMapState struct {
+	Action           string `json:"action"`
+	OnMapExpire      string `json:"on_map_expire"`
+	OnActionExecuted string `json:"on_action_executed"`
+}
+
 type UpdateControlMap struct {
-	ID       string `json:"id"`
-	Priority int    `json:"priority"`
+	ID       string                           `json:"id"`
+	Priority int                              `json:"priority"`
+	States   map[string]UpdateControlMapState `json:"states"`
+}
+
+var UpdateControlMapStateKeyValid []string = []string{
```

Comment:
Nit:
const

Here and below

Reply:
You cannot have const arrays/slice in golang

Reply:
:exploding_head: TIL